censys/cli.py

```diff
@@ -0,0 +1,685 @@
+            "443.https.tls.certificate.parsed.extensions.subject_alt_name.dns_names",
+        ]
+
+        query = kwargs.get("query", "")
+        fields = kwargs.get("fields", [])
+        overwrite = kwargs.get("overwrite", False)
+
+        index = CensysIPv4(api_id=self.api_user, api_secret=self.api_pass)
+
+        return self._process_search(
+            query,
+            index,
+            self._combine_fields(default_fields, fields, overwrite=overwrite),
+        )
+
+    def search_certificates(self, **kwargs) -> Results:
+        """
+        A method to search the Certificates data set via the API.
+
+        Args:
+            kwargs:
+                query: The string search query.
+                fields: The fields that should be returned with a query.
+                overwrite: Overwrite the default list of fields with the given fields.
+
+        Returns:
+            Results: A list of search results from an API query.
+        """
+
+        default_fields = [
+            "metadata.updated_at",
+            "parsed.issuer.common_name",
+            "parsed.names",
+            "parsed.serial_number",
+            "parsed.self_signed",
+            "parsed.subject.common_name",
+            "parsed.validity.start",
+            "parsed.validity.end",
+            "parsed.validity.length",
+            "metadata.source",
+            "metadata.seen_in_scan",
+            "tags",
+        ]
+
+        query = kwargs.get("query", "")
+        fields = kwargs.get("fields", [])
+        overwrite = kwargs.get("overwrite", False)
+
+        index = CensysCertificates(api_id=self.api_user, api_secret=self.api_pass)
+
+        return self._process_search(
+            query,
+            index,
+            self._combine_fields(default_fields, fields, overwrite=overwrite),
+        )
+
+    def search_websites(self, **kwargs) -> Results:
+        """
+        A method to search the Websites (Alexa Top 1M) data set via the API.
+
+        Args:
+            kwargs:
+                query: The string search query.
+                fields: The fields that should be returned with a query.
+                overwrite: Overwrite the default list of fields with the given fields.
+
+        Returns:
+            Results: A list of search results from an API query.
+        """
+
+        default_fields = [
+            "443.https.tls.version",
+            "alexa_rank",
+            "domain",
+            "ports",
+            "protocols",
+            "tags",
+            "updated_at",
+        ]
+
+        query = kwargs.get("query", "")
+        fields = kwargs.get("fields", [])
+        overwrite = kwargs.get("overwrite", False)
+
+        index = CensysWebsites(api_id=self.api_user, api_secret=self.api_pass)
+
+        return self._process_search(
+            query,
+            index,
+            self._combine_fields(default_fields, fields, overwrite=overwrite),
+        )
+
+
+class CensysHNRI:
```

Comment:
Is this exactly what HNRI does for risk computation? If not - it might create some confusion on the HNRI product.

Reply:
It uses the exact same code from me.censys.io. Should I rename it to `me`?

Reply:
After some outside discussion, HNRI does make sense here (it's what we call it on me.censys.io) - the "enhanced" version we offer through the SAS product can be seen as sufficiently distinct as the tools offered here are more "DIY".

Reply:
Sounds good 👍